app/code/Magento/InventoryApi/Test/Api/SourceRepository/UpdateTest.php

```diff
@@ -0,0 +1,129 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+namespace Magento\InventoryApi\Test\Api\SourceRepository;
+
+use Magento\Framework\Webapi\Rest\Request;
+use Magento\InventoryApi\Api\Data\SourceCarrierLinkInterface;
+use Magento\InventoryApi\Api\Data\SourceInterface;
+use Magento\TestFramework\Assert\AssertArrayContains;
+use Magento\TestFramework\TestCase\WebapiAbstract;
+
+class UpdateTest extends WebapiAbstract
+{
+    /**#@+
+     * Service constants
+     */
+    const SOURCE_PATH = '/V1/inventory/source';
+    const SOURCES_PATH = '/V1/inventory/sources';
+    const SERVICE_NAME = 'inventorySourceRepositoryV1';
+    /**#@-*/
+
+    /**
+     * @magentoApiDataFixture ../../../../app/code/Magento/InventoryApi/Test/_files/source/source_1.php
+     */
+    public function testUpdate()
+    {
+        $source = $this->getSourceDataByName('source-name-1');
+        $sourceId = $source[SourceInterface::SOURCE_ID];
+        $data = [
+            SourceInterface::NAME => 'source-name-1-updated',
+            SourceInterface::CONTACT_NAME => 'source-contact-name-1-updated',
+            SourceInterface::EMAIL => 'source-email-1-updated',
+            SourceInterface::ENABLED => false,
+            SourceInterface::DESCRIPTION => 'source-description-1-updated',
+            SourceInterface::LATITUDE => 0.3,
+            SourceInterface::LONGITUDE => 0.2,
```

Comment:
to make the test more realistic please use more digits after the dot for longitude and latitude

like, `12.1234567`